File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
	
	k %= nums.size();
    reverse(nums.begin(), nums.begin() + nums.size() - k);
    reverse(nums.rbegin(),nums.rbegin() + k);
    reverse(nums.begin(), nums.end());
}
};

File: 0189-rotate-array/0189-rotate-array.java
class Solution {
   public void rotate(int[] nums, int k) {
        Deque<Integer> deque = new ArrayDeque<>();
        for(int a : nums)
            deque.add(a);
        k %= nums.length;
        while(k-- != 0){
           deque.addFirst(deque.removeLast());
        }
        for(int i=0;i<nums.length;i++){
            nums[i] = deque.removeFirst();
        }
    }
}